cmd/jujud/updateseries/updateseries.go

```diff
@@ -118,37 +138,58 @@ func (c *UpdateSeriesCommand) Run(ctx *cmd.Context) error {
 		return err
 	}
 
+	jujuVersion, err := c.getJujuVersion()
+	if err != nil {
+		return err
+	}
+
 	switch {
 	case toInitSys == service.InitSystemUpstart:
 		return errors.NewNotSupported(nil, "downgrade to series using upstart not supported")
 	case toInitSys == fromInitSys && toInitSys == service.InitSystemSystemd:
-		if err = c.copyTools(); err != nil {
+		if err = service.CopyAgentBinary(c.machineAgent, c.unitAgents, c.dataDir, c.toSeries, c.fromSeries, jujuVersion); err != nil {
 			return err
 		} else {
 			ctx.Infof("successfully copied tools and relinked agent tools")
 		}
 		if !c.startAgents {
 			break
 		}
-		if err = c.startAllAgents(ctx); err != nil {
+		if errMachineName, errUnitNames, err = service.StartAllAgents(c.machineAgent, c.unitAgents, c.dataDir, c.toSeries); err != nil {
 			return err
+		} else {
+			for _, unitName := range errUnitNames {
```

Comment:
errUnitNames doesn't seem right?
Should it be startedUnitNames?
Same with machine

Actually, there's no need to distinguish between machine and unit.
Just return a single slice of started names. And the slice will have "jujud-foo" in it rather than the caller knowing to have to add "jujud-" as a prefix.